src/vivarium_nih_us_cvd/components/treatment.py

```diff
@@ -700,26 +738,54 @@ def enroll_in_outreach(
         new_outreach = self.outreach(maybe_enroll)
         to_enroll = current_outreach[current_outreach != new_outreach].index
         if not to_enroll.empty:
-            # Update the outreach column with the new pipeline values. This
-            # is then used in OutreachEffect which registers a value modifier
-            # using the (newly updated) outreach column to modify adherence exposure.
+            # Update the outreach column with pipeline values
             pop_visitors.loc[to_enroll, data_values.COLUMNS.OUTREACH] = new_outreach.loc[
                 to_enroll
             ]
-            self.population_view.update(pop_visitors[[data_values.COLUMNS.OUTREACH]])
-            # With the just-updated outreach column, update the medication
-            # adherence columns with pipeline values
+            # Update the medication adherence columns with pipeline values
             pop_visitors.loc[
                 to_enroll, data_values.COLUMNS.SBP_MEDICATION_ADHERENCE
             ] = self.sbp_medication_adherence(to_enroll)
             pop_visitors.loc[
                 to_enroll, data_values.COLUMNS.LDLC_MEDICATION_ADHERENCE
             ] = self.ldlc_medication_adherence(to_enroll)
+
             self.population_view.update(
                 pop_visitors[
                     [
                         data_values.COLUMNS.SBP_MEDICATION_ADHERENCE,
                         data_values.COLUMNS.LDLC_MEDICATION_ADHERENCE,
+                        data_values.COLUMNS.OUTREACH,
+                    ]
+                ]
+            )
+
+        return pop_visitors
+
+    def enroll_in_polypill(
+        self, pop_visitors: pd.DataFrame, maybe_enroll: pd.Index
+    ) -> pd.DataFrame:
+        """Enrolls simulants in polypill intervention program. It updates
+        the polypill column as well as sbp medication adherence column (the
+        effect of polypill intervention).
+
+        Reminder: 'cat1' polypill means enrolled and 'cat2' means not enrolled.
+        """
+        current = pop_visitors.loc[maybe_enroll, data_values.COLUMNS.POLYPILL]
```

Comment:
You can replace all of this with:

```
pop_visitors.loc[maybe_enroll, data_values.COLUMNS.POLYPILL] = self.polypill(maybe_enroll)
pop_visitors.loc[maybe_enroll, data_values.COLUMNS.SBP_MEDICATION_ADHERENCE] = self.sbp_medication_adherence(maybe_enroll)
```

This is because of the way polypill acts on the exposure parameters of adherence and not the exposure itself, as we discussed this morning, so it isn't incorrect to call it on people who are already on the intervention.